src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractDatasetCommand.java

```diff
@@ -221,4 +223,79 @@ public void setDataset(Dataset dataset) {
     protected Timestamp getTimestamp() {
         return timestamp;
     }
+
+    protected void registerFilePidsIfNeeded(Dataset theDataset, CommandContext ctxt, boolean b) throws CommandException {
+     // Register file PIDs if needed
+        String protocol = theDataset.getProtocol();
+        String authority = theDataset.getAuthority();
+        GlobalIdServiceBean idServiceBean = GlobalIdServiceBean.getBean(protocol, ctxt);
+        String currentGlobalIdProtocol = ctxt.settings().getValueForKey(SettingsServiceBean.Key.Protocol, "");
+        String currentGlobalAuthority = ctxt.settings().getValueForKey(SettingsServiceBean.Key.Authority, "");
+        String dataFilePIDFormat = ctxt.settings().getValueForKey(SettingsServiceBean.Key.DataFilePIDFormat, "DEPENDENT");
+        boolean shouldRegister = ctxt.systemConfig().isFilePIDsEnabled()                                  // We use file PIDs
+                && !idServiceBean.registerWhenPublished()                                                 // The provider can pre-register
+                &&((currentGlobalIdProtocol.equals(protocol) && currentGlobalAuthority.equals(authority)) // the dataset PID is a protocol/authority Dataverse can create new PIDs in
+                        || dataFilePIDFormat.equals("INDEPENDENT"));                                      // or the files can use a different protocol/authority
+        logger.fine("IsFilePIDsEnabled: " + ctxt.systemConfig().isFilePIDsEnabled());
+        logger.fine("RegWhenPub: " +  !idServiceBean.registerWhenPublished());
+        logger.fine("OK provider: " + ((currentGlobalIdProtocol.equals(protocol) && currentGlobalAuthority.equals(authority)) // the dataset PID is a protocol/authority Dataverse can create new PIDs in
+                        || dataFilePIDFormat.equals("INDEPENDENT")));  
```

Comment:
I know it's more code but rather than having hardcoded strings like "INDEPENDENT" my preference is to reference the place where the string comes from like this: SystemConfig.DataFilePIDFormat.INDEPENDENT.toString()

That way if you do a "find usages" in your editor, it easily comes up. And if we ever change the string (which is doubtful), it's easier to find.

This might just be a me think. Please feel free to get opinions from others.